src/NUnitEngine/nunit.engine/Internal/ServerUtilities.cs

```diff
@@ -36,114 +37,104 @@ namespace NUnit.Engine.Internal
     /// </summary>
     public static class ServerUtilities
     {
-        static Logger log = InternalTrace.GetLogger(typeof(ServerUtilities));
+        private static readonly Logger Log = InternalTrace.GetLogger(typeof(ServerUtilities));
 
         /// <summary>
-        ///  Create a <see cref="TcpChannel"/> with a given name on a given port.
+        /// Create a <see cref="TcpChannel"/> with a given name on a given port.
         /// </summary>
         /// <param name="name">The name of the channel to create.</param>
         /// <param name="port">The port number of the channel to create.</param>
         /// <param name="limit">The rate limit of the channel to create.</param>
         /// <returns>A <see cref="TcpChannel"/> configured with the given name and port.</returns>
-        private static TcpChannel CreateTcpChannel( string name, int port, int limit )
+        private static TcpChannel CreateTcpChannel(string name, int port, int limit)
         {
-            Hashtable props = new Hashtable();
-            props.Add( "port", port );
-            props.Add( "name", name );
-            props.Add( "bindTo", "127.0.0.1" );
-
-            BinaryServerFormatterSinkProvider serverProvider =
-                new BinaryServerFormatterSinkProvider();
-
-            // NOTE: TypeFilterLevel and "clientConnectionLimit" property don't exist in .NET 1.0.
-            Type typeFilterLevelType = typeof(object).Assembly.GetType("System.Runtime.Serialization.Formatters.TypeFilterLevel");
-            if (typeFilterLevelType != null)
+            var props = new Dictionary<string, object>
             {
-                PropertyInfo typeFilterLevelProperty = serverProvider.GetType().GetProperty("TypeFilterLevel");
-                object typeFilterLevel = Enum.Parse(typeFilterLevelType, "Full");
-                typeFilterLevelProperty.SetValue(serverProvider, typeFilterLevel, null);
+                { "port", port },
+                { "name", name },
+                { "bindTo", "127.0.0.1" },
+                //{ "clientConnectionLimit", limit }
+            };
 
-//                props.Add("clientConnectionLimit", limit);
-            }
+            var serverProvider = new BinaryServerFormatterSinkProvider
+            {
+                TypeFilterLevel = TypeFilterLevel.Full
+            };
 
-            BinaryClientFormatterSinkProvider clientProvider =
-                new BinaryClientFormatterSinkProvider();
+            var clientProvider = new BinaryClientFormatterSinkProvider();
 
-            return new TcpChannel( props, clientProvider, serverProvider );
+            return new TcpChannel(props, clientProvider, serverProvider);
```

Comment:
All looks good. Like @ChrisMaddock says, nice to see the .NET 1.0 hack removed, the code is much cleaner. I'm still not sure why we need a `TcpChannel` here where an `IpcChannel` would do and would likely be quicker. Not really part of this PR, just thinking out loud 😄